api/scaler/dispatcher.go

```diff
@@ -0,0 +1,78 @@
+package scaler
+
+import (
+	"log"
+	"time"
+
+	"github.com/quintilesims/layer0/api/provider"
+)
+
+const (
+	SCALE_GRACE_PERIOD = time.Second * 15
+)
+
+var (
+	timeMultiplier time.Duration = 1
+)
+
+type Dispatcher struct {
+	environmentProvider provider.EnvironmentProvider
+	scaler              Scaler
+	schedule            map[string]*time.Timer
+}
+
+func NewDispatcher(e provider.EnvironmentProvider, s Scaler) *Dispatcher {
+	return &Dispatcher{
+		environmentProvider: e,
+		scaler:              s,
+		schedule:            map[string]*time.Timer{},
+	}
+}
+
+func (d *Dispatcher) ScheduleRun(environmentID string) {
+	if timer, ok := d.schedule[environmentID]; ok {
```

Comment:
I feel like we're mixing our terminology with AWS' terminology a bit regard nouns and verbs. AWS uses the term 'scale' when applied to `cluster`s, but from a user/client perspective, `service`s are the entity that is scaled, not `environment`. `Environment` instead uses `size`, (which may or may not be "correct").

Reply:
hmm, I'm not sure what a better alternative to "scale an environment" would be though?

The action we want is to get the exact right amount of ec2 instances in a cluster. Calibrate?

Reply:
It seems like, in general, we make things more confusing by attempting to rebrand AWS terms. `Scale` makes a lot of sense here because that's exactly what's happening AND it coincides with AWS' terminology. We should consider using a new verb for `service`s --- possibly just `AutoScale` to properly differentiate. This will probably have more implications for when we finally get around to implementing Application Load Balancers